### DISCUSSION TITLE
Howto - Ensure consistency in source language?

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#6369] [Q&A]
Concrete case: I want to make sure that 'log file' is always spelt this way and never 'logfile' or 'log-file'.

My idea was to add the two wrong spellings to the glossary and flag them as forbidden terms. I then used mass edit to set the 'check-glossary' flag for all strings.

This kind of works (first screenshot), since it triggers the "does not follow glossary" warning, but it does this for all languages (second screenshot), where I'd like it to only check the English source.

![grafik](https://user-images.githubusercontent.com/3872576/127861003-86dae193-eb8a-4eab-b678-d6ccdab18693.png)
![grafik](https://user-images.githubusercontent.com/3872576/127861280-1c577d9c-b94b-4124-b3e9-b637a01a91fb.png)

Is there anything similar to https://docs.weblate.org/en/latest/user/checks.html#does-not-follow-glossary, but for the source language only?

**Comment by nijel** [comment]
> My idea was to add the two wrong spellings to the glossary and flag them as forbidden terms. I then used mass edit to set the 'check-glossary' flag for all strings.

The better idea is to set this at the component level instead of editing flag of all strings :-)

Back to your question - I don't thing Weblate has a solution for this right now.

**Reply by luebbe** [reply]
> The better idea is to set this at the component level instead of editing flag of all strings :-)

Yup, I noticed that after I wrote this question, changed our config accordingly, but forgot to update the Q&A here.
 
> Back to your question - I don't thing Weblate has a solution for this right now.

Would that be worth a feature request? I see it as a valid use case.

**Reply by nijel** [reply]
I see a valid use case, but I'm not confident that glossary is the right place to address this.

**Reply by luebbe** [reply]
You're right. Spelling of the source language should be correct and terminology consistent before this content makes it into Weblate. OTOH Weblate is (at least for me) the only instance, where everything comes together, so using and maybe abusing the "forbidden-terms" glossary was the first thing that came to mind.

**Reply by nijel** [reply]
Weblate does have source string checks, so it's in scope, I just don't think polluting glossary with these is a good idea (it will confuse translators).
